File: iOS7bookExamples/bk2ch06p275tabbedInterface/ch19p611tabbedInterface/FirstViewController.m


#import "FirstViewController.h"

@interface FirstViewController () <UITabBarControllerDelegate>

@end

@implementation FirstViewController

-(void)viewDidLoad {
    [super viewDidLoad];
    self.tabBarController.delegate = self; // new iOS 7 delegate feature...
    // delegate gets to dictate rotation rules
}

-(NSUInteger)tabBarControllerSupportedInterfaceOrientations: (UITabBarController *) tabBarController {
    return UIInterfaceOrientationMaskPortrait;
}

-(NSUInteger)supportedInterfaceOrientations {
    return UIInterfaceOrientationMaskPortrait; // pointless
}


@end

File: bk2ch06p275tabbedInterface/ch19p611tabbedInterface/FirstViewController.swift


import UIKit


class FirstViewController : UIViewController, UITabBarControllerDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.tabBarController?.delegate = self
    }
    
    // tab bar delegate gets to dictate orientations...
    // so there's no need to subclass UITabBarController
    
    func tabBarControllerSupportedInterfaceOrientations(_ tabBarController: UITabBarController) -> UIInterfaceOrientationMask {
        print(self, terminator: "")
        print(" ", terminator: "")
        print(#function)
        return .portrait    }
    
    override var supportedInterfaceOrientations : UIInterfaceOrientationMask {
        print(self, terminator: "")
        print(" ", terminator: "")
        print(#function)
        return .landscape // called, but pointless
    }
    
    override var prefersStatusBarHidden : Bool {
        return true
    }
    

    
}
